tests/test_array.py

```diff
@@ -1146,6 +1171,50 @@ async def test_v2_chunk_encoding(
         assert arr.compressors == compressor_expected
         assert arr.filters == filters_expected
 
+    @staticmethod
+    async def test_invalid_chunk_encoding(store: MemoryStore) -> None:
+        """
+        Test that passing an invalid compressor or filter to create_array raises an error.
+        """
+        invalid_compressor_type = 2
+        msg = f"For Zarr format 2 arrays, the `compressor` must be a single codec. Expected None, a numcodecs.abc.Codec, or a dict or str representation of a numcodecs.abc.Codec. Got {type(invalid_compressor_type)} instead."
+        with pytest.raises(ValueError, match=msg):
+            await create_array(
+                store=store,
+                dtype="uint8",
+                shape=(10,),
+                zarr_format=2,
+                compressors=invalid_compressor_type,
+            )
+        with pytest.raises(KeyError):
+            await create_array(
+                store=store,
+                dtype="uint8",
+                shape=(10,),
+                zarr_format=3,
+                filters="nonexistent_filter_name",
+            )
+
+        # string representation of a codec is only supported if codec has no required arguments
+        msg = "Delta.__init__() missing 1 required positional argument: 'dtype'"
+        with pytest.raises(TypeError, match=re.escape(msg)):
+            await create_array(
+                store=store,
+                dtype="uint8",
+                shape=(10,),
+                zarr_format=2,
+                filters="delta",
+            )
+        msg = "TransposeCodec.__init__() missing 1 required keyword-only argument: 'order'"
```

Comment:
i don't think this kind of error message is the best outcome here. it would be better to have an error like "you provided the string foo, which specifies codec fooCodec, but that codec cannot be specified by a string because it takes a required configuration. use either the dict form of fooCodec, or pass in a concrete fooCodec instance instead"